### ISSUE TITLE
document RDF generation and the LUBM ontology

### ISSUE THREAD

**Issue by VladimirAlexiev** [LiUGraphQL/LinGBM#83]
I assume this test targets GraphQL implementations over RDF (like the Ontotext Platform).
- at https://github.com/LiUGraphQL/LinGBM/tree/master/tools/datasetgen please describe how to generate RDF vs relational data (eg using `--format TURTLE`)
- document where is the LUBM ontology
- Change https://github.com/LiUGraphQL/LinGBM/wiki/Mapping-of-GraphQL-Schema-to-Relational-Schema-TODO to cover the mapping of both the relational schema and the ontology. I strongly hope that the two are isomorphic, so can be documented together

**Comment by hartig** [comment]
All these things sound good and should be done. However, just to scale down the expectations in us: The reality is that we currently have one PhD student (@chengsijin0817) who can work **some part** of her time on LinGBM, and I can advise and moderate a bit. So, we can focus only on specific pieces out of the many directions into which this benchmark project can be taken. If we are lucky, we may find a Master's student from time to time who may help by working on a smaller project within the context of LinGBM (properly extending LinGBM to RDF-based data sources would be one such project, but we don't have such a student at the moment). So, any external help we can get is appreciated.

**Comment by VladimirAlexiev** [comment]
@hartig running this over RDF is not yet in our sprint plans, but will likely be included. So we'll extend it to RDF.

You changed the LUBM gen to produce RDBMS data, is it isomorphic to the original RDF?

**Comment by hartig** [comment]
Yes, the RDBMS data produced by our dataset generator is isomorphic to the RDF data. Our extension to the dataset generator intercepts the produced RDF data and turns it into SQL DML statements (INSERT INTO and UPDATE) for an SQL import script. The actual data is _not_ changed in the process. (@chengsijin0817 can you confirm?)

**Comment by hartig** [comment]
After discussing the question of isomorphism internally, I have to backtrack a bit: While it is indeed true that our extension to the dataset generator intercepts the produced RDF data in order to turn it into SQL DML statements, we actually also add a bit of additionally data; namely, for each publication, we add a title and abstract, and for each student, we add an age. These additional attributes allow us to cover some of our choke points that we could not cover based on the original LUBM data. Currently, these additional attributes are present only in the RDBMS data that our extended dataset generator produces. We can check whether we can move the generation of these attributes deeper into the dataset generator such that they also end up as additional datatype properties in the generated RDF data.

**Comment by VladimirAlexiev** [comment]
Extending LUBM with more choke points is a great idea. 

Does anyone maintain LUBM, or will you he taking that role?

**Comment by hartig** [comment]
> Does anyone maintain LUBM

I don't know. 

> or will you he taking that role?

No time.
